source/extensions/common/aws/signer_impl.cc

```diff
@@ -76,7 +77,6 @@ std::string SignerImpl::createContentHash(Http::Message& message, bool sign_body
   const auto content_hash = message.body()
                                 ? Hex::encode(crypto_util.getSha256Digest(*message.body()))
                                 : SignatureConstants::get().HashedEmptyString;
-  message.headers().addCopy(SignatureHeaders::get().ContentSha256, content_hash);
```

Comment:
Shouldn't this method take a `const Http::Message&` now? Im surprised this method mutates the message tbh.

Reply:
It's always been like that.